amundsen_application/api/utils/search_utils.py

```diff
@@ -51,3 +53,9 @@ def generate_query_json(*, filters: Dict = {}, page_index: int, search_term: str
         },
         'query_term': search_term
     }
+
+
+def _map_badges(badges: List[str]) -> List[Dict[str, str]]:
+    if not badges:
+        return []
+    return [{"tag_name": badge, "tag_type": "badge"} for badge in badges]
```

Comment:
I did this because the metadata service returns tags as tags: [{tag_name: name, tag_type: type}] and it doesn't make sense to have the front end handle the mapping. 
I wasn't sure if it makes more sense to make a model here or not given how other results are handled.

Reply:
Did you mean the metadata service or the search service? If metadata service wondering how that applies to this feature. If search service, what are the repercussions using the information directly? 

Overall I would expect that the search service can return exactly what the frontend service needs. This is not currently always the case with some features as you may have noticed, but I think it's a problem that should more actively addressed in new features vs accumulating more workarounds.

Reply:
I meant the metadata service - it's the only other entity returning badge information that is consumed by the front end. The reference to it was with regards to keeping badge data output consistent so that the FE doesn't have to worry about it. 

The search service already doesn't return exactly what the front end needs, but more a direct mapping of the search providers output (ie from ES/Atlas) to a table or user entity as defined by the search services models. The mapping of a badge model to what the front end expects seems more like an implementation detail for the front end service than for the search service from my perspective. 

But I'm happy to change things if it makes more sense to make the mapping further upstream.

Reply:
I'd prefer the mapping to be further upstream, and for amundsensearch to represent a badge using the full object that represents it. So far we've been lucky that all the values we've needed to show in the search results were strings -- like table names and descriptions -- so badges introduces a new case that should be handled differently if possible.

There is a need for the frontend to have the full representation of the badge object in the search results. While it's easy to throw in the `tag_type` hardcoded as "badge" right now, I'm not sure what we would do if and when we extend the badge object. For example, introducing a `category` value into badges is the only scalable way for the frontend to know what color different badges should be so that we can do a mapping from category to color.

Reply:
IMO the concept of badge is not e2e 100% well designed/decided. In UI, it is not editable , but we never consider how we ingest the badge and who could ingest the badge. And the data model is 100% exactly the same as tag which it exactly fits into the tag model to avoid duplicate code (otherwise we will have so many duplicate code in metadata service/search).

Reply:
but the change here is a bit hacky...I think we need to think through on how to allow FE to understand the badge concept without too much code by handling tag vs badge in the search service...

Reply:
I was working on making the search service and databuilder handle tag vs badge per tamika's comments - though I feel like doing it at the databuilder level seems not great.